src/gui/notificationService/NotificationServiceFactory.cpp

```diff
@@ -18,16 +18,20 @@
  */
 
 #include "NotificationServiceFactory.h"
+#include "src/backend/config/KsnipConfigProvider.h"
 
 #if defined(UNIX_X11)
 #include "FreeDesktopNotificationService.h"
 #endif
 
 IToastService *NotificationServiceFactory::create(IToastService *defaultNotificationService)
 {
+	if (!KsnipConfigProvider::instance()->platformSpecificNotificationServiceEnabled()) {
+		return defaultNotificationService;
+	} else {
 #if defined(UNIX_X11)
```

Comment:
It's fine when you duplicate code here. You can have the if defined(UNIX_X11) and in it the if/else to check if to return default or freeDesktop. Then you can have the else of the defined with a single return. In that way I thinbk it's more readable.